source/state_representation/src/State.cpp

```diff
@@ -33,12 +33,14 @@ void State::set_filled() {
   this->reset_timestamp();
```

Comment:
can you add a comment here to remember the next PR?
```suggestion
  # FIXME: resetting the timestamp should be done explicitly, not within set_filled
  this->reset_timestamp();
```